lib/saluki-components/src/sources/checks/mod.rs

```diff
@@ -60,12 +59,21 @@ pub struct ChecksConfiguration {
     /// Autodiscovery provider to use.
     #[serde(skip)]
     autodiscovery_provider: Option<Arc<dyn AutodiscoveryProvider + Send + Sync>>,
+
+    /// Hostname provider to use.
+    #[serde(skip)]
+    hostname_provider: Option<Arc<dyn HostProvider<Error = GenericError> + Send + Sync>>,
```

Comment:
Since you're now holding two distinct providers, it might be a little cleaner to just hold the environment provider instead. You can look at the Host Enrichment transform for an example of how we do that, including where to place the necessary bounds on the `SourceBuilder` implementation for stuff like the error type, etc.

Reply:
Gave it a go here https://github.com/DataDog/saluki/pull/787/commits/abd50d4facf511d2753a2a51e353cecee3f2180a